dataRetriever/dataPool/proofsCache/proofsCache.go

```diff
@@ -14,16 +13,17 @@ type proofNonceMapping struct {
 }
 
 type proofsCache struct {
-	mutProofsCache sync.RWMutex
-	proofsByNonce  []*proofNonceMapping
-	proofsByHash   map[string]data.HeaderProofHandler
+	mutProofsCache       sync.RWMutex
+	proofsByNonceBuckets []*proofNonceBucket
```

Comment:
I think this can be a map as well with the key the lowest nonce that the bucket would hold

Reply:
changed to use `sync.Map`